Added WithErrorCode Method to pass Custom Error Code #150

vikashajela opened the pull request:
Hi Jeremy,

As discussed we have added a new method (i.e. WithErrorCode) to pass custom error code. With this method Custom/user-defined Error Code can passed both independently or along with Error Message. 
As suggested by you we are using the existing ErrorCode Property (in ValidationFailure.cs) to persist the Custom ErrorCode passed by users.

[Test Cases] 
We have executed all the existing test cases and they are passing successfully. 

We have also added two new test cases considering following scenarios:

1. Passing Just ErrorCode in WithErrorCode method.
2. Passing both Error Code and ErrorMessage together in a chaining with methods WithErrorCode & WithErrorMessge respectively.    
   
[Documentation]

We have also created the documentation for the newly added method, pls find it at below link. 

https://docs.google.com/document/d/1B0N3qXErBp3SfvSiB3xWYryTh0FH0xVAzPxyCleCP3o/edit?usp=sharing


Please review all the work done by us and let us know for any corrections or further changes.


Regards,

Vikas

JeremySkinner commented:
Thanks for this - please could you reformat the changes to use tabs instead of spaces, for consistency with the rest of the project? I'll then get this merged in. 

Thanks

vikashajela commented:
Hi Jeremy,

We have replaced the spaces with tabs in our branch, request you to pls now pull the code.
We have also modified the Wiki documentation in our branch, request you to pls pull that also.

Please let us know if there are any more issues.

Regards,

Vikas

vikashajela commented:
Hi Jeremy,

I can find that our pull request is still open. Please let us know if you need anything more from us on this.

Regards,

Vikas

JeremySkinner commented:
Thanks, I'm aiming to get this merged in this week.

vikashajela commented:
Thanks Jeremy for merging this.

vikashajela commented:
Thanks Jeremy for Merging, pls let us know if we can help you in any more
issues.


Regards,

Vikas

On Mon, Nov 30, 2015 at 9:35 PM, Jeremy Skinner <notifications@github.com>
wrote:

> Merged #163 <https://github.com/JeremySkinner/FluentValidation/pull/163>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JeremySkinner/FluentValidation/pull/163#event-477880675>
> .
>
